### ISSUE TITLE
New Charts.js shows blank spaces when there is no data

### ISSUE THREAD

**Issue by pxpto** [home-assistant/home-assistant-polymer#1014]
This didn't happen before with Google Charts:

![image](https://user-images.githubusercontent.com/5114388/37727175-0ce3a416-2d2f-11e8-994f-5d0f88dbd812.png)

**Comment by NovapaX** [comment]
Apparently there were states, but they had no value.... Probably because the sensor was unavailable?
Then this is what I would expect. We should not make up our own history.

**Comment by pxpto** [comment]
I agree we shouldn't make up our own history. But leaving blank spaces in charts?! This didn't happened before with Google Charts.

**Comment by NovapaX** [comment]
What should it display then? If it draws a line there it 'overrides' the history at that point in time which apparently has a no-value state.
Maybe a greyed out area to show that this is no error but indeed no values.

What state does the sensor report in those no-line areas? Is it 'unavailable'

**Comment by balloob** [comment]
I agree with @NovapaX. We should not show any data if we don't have any. Just taking the average between 2 points cannot be assumed to be always correct

> This didn't happened before with Google Charts.

It is not our goal to re-implement Google Charts.

**Comment by fanthos** [comment]
This is intended behavior. If the data has `null`, `undefined` or `unavailable`, the line will shows blank until next usable data.
One chart might has multiple lines, so not sure its good idea to use grayed out area.
